spec/lexers/ocaml_spec.rb

```diff
@@ -17,5 +17,31 @@
       assert_guess :mimetype => 'text/x-ocaml'
     end
   end
+
+  describe 'lexing' do
+    include Support::Lexing
+
+    it 'recognizes value as a Name' do
+      assert_tokens_equal 'value', ['Name', 'value']
+    end
+
+    it 'recognizes nonrec as a Keyword' do
+      assert_tokens_equal 'nonrec', ['Keyword', 'nonrec']
+    end
+
+    it 'recognizes local open expressions' do
+      assert_no_errors 'M.N.(f x)'
+      assert_no_errors 'M.N.{f = 0}'
+      assert_no_errors 'M.N.[0]'
+    end
+
+    it 'recognizes polymorphic variants' do
+      assert_no_errors '`max'
+    end
+
+    it 'recognizes identifiers that start with _' do
+      assert_no_errors 'let x = M._x'
+    end
+  end
```

Comment:
done. is there a way to automatically check that those visual samples do not have errors?